src/JitInterface/src/CorInfoImpl.Intrinsics.cs

```diff
@@ -9,7 +9,7 @@
 
 namespace Internal.JitInterface
 {
-    internal unsafe partial class CorInfoImpl
+    public unsafe partial class CorInfoImpl
```

Comment:
What composition we're going for, with the refactoring in #5985 in mind?

My line of thinking was:

```
                         ILCompiler.Compiler
                 ________^        ^        ^___    
                /                 |            \
ILCompiler.RyuJIT       ILCompiler.ReadyToRun  ILCompiler.CppCodegen
  Targets CoreRT           Targets CoreCLR        Targets CoreRT
private CorInfoImpl      private CorInfoImpl
```

If it aligns with what you want to do, maybe instead of making all of these `public`, we should just include the files in the ReadyToRun project so that it doesn't immediately break once we integrate #5985 that removes them from ILCompiler.Compiler.

We could eventually also think about splitting the CoreRT-specific things out of ILCompiler.Compiler into ILCompiler.CoreRT (that the individual codegen backends targeting CoreRT would depend on). But that's a nice to have for the future. (I like to have these kinds of things because it helps people new to the project to focus on what's important for the thing they want to do, instead of getting distracted by things that are not applicable to the runtime they're targeting anyway - and it helps us in seeing how far away these are from each other),

Reply:
> (I like to have these kinds of things because it helps people new to the project to focus on what's important for the thing they want to do, instead of getting distracted by things that are not applicable to the runtime they're targeting anyway - and it helps us in seeing how far away these are from each other),

For-what-its-worth, I agree. 

When I was writing my [blog post on CoreRT](http://mattwarren.org/2018/06/07/CoreRT-.NET-Runtime-for-AOT/#compiler) it confused me why `RyuJITCompilation` lived in the common part of the compiler (i.e. under `ILCompiler.Compiler`)